climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,891 @@
+    geometries to points.
+
+    Parameters
+    ----------
+    exp : Exposures
+        The exposure instance with exp.gdf.geometry containing (multi-)polygons
+    res : float
+        Resolution of the disaggregation grid.
+    to_meters : bool, optional
+       If True, the geometries are projected to an equal area projection before
+       the disaggregation. res is then in meters. The exposures are
+       then reprojected into the original projections before the impact
+       calculation. The default is False.
+    disagg : string, optional
+        Disaggregation method.
+        if 'avg' value is average over points
+        if 'surf' value is surface per points. Distance per point (res) for lines.
+        if 'None' value is unchanged (or 1 if no value is defined)
+
+    Returns
+    -------
+    exp_pnt : Exposures
+        Exposures with a double index geodataframe, first for the geometries of exp,
+        second for the point disaggregation of the geometries.
+
+    """
+
+    # rasterize
+    if to_meters:
+        gdf_pnt = line_to_pnts_m(gdf, res)
+    else:
+        gdf_pnt = line_to_pnts(gdf, res)
+
+    # disaggregate
+    if disagg == 'avg':
+        gdf_pnt = disagg_gdf_avg(gdf_pnt)
+    elif disagg == 'surf':
+        gdf_pnt = disagg_gdf_val(gdf_pnt, res)
+    elif disagg is None and 'value' not in gdf_pnt.columns:
+        gdf_pnt['value'] = 1
+
+    return gdf_pnt
+
+
+def _gdf_poly_to_pnt(gdf, res, to_meters, disagg):
+    """
+    Disaggregate exposures with (multi-)polygons
+    geometries to points.
+
+    Parameters
+    ----------
+    exp : Exposures
+        The exposure instance with exp.gdf.geometry containing (multi-)polygons
+    res : float
+        Resolution of the disaggregation grid.
+    to_meters : bool
+       If True, the geometries are projected to an equal area projection before
+       the disaggregation. res is then in meters. The exposures are
+       then reprojected into the original projections before the impact
+       calculation.
+    disagg : string
+        Disaggregation method.
+        if 'avg' value is average over points
+        if 'surf' value is surface per points. Area per point (res*res)
+            for polygons.
+        if 'None' value is unchanged (or 1 if no value is defined)
+
+    Returns
+    -------
+    exp_pnt : Exposures
+        Exposures with a double index geodataframe, first for the geometries of exp,
+        second for the point disaggregation of the geometries.
+
+    """
+
+    # rasterize
+    if to_meters:
+        gdf_pnt = poly_to_pnts_m(gdf, res)
+    else:
+        gdf_pnt = poly_to_pnts(gdf, res)
+
+    # disaggregate
+    if disagg == 'avg':
+        gdf_pnt = disagg_gdf_avg(gdf_pnt)
+    elif disagg == 'surf':
+        gdf_pnt = disagg_gdf_val(gdf_pnt, res * res)
+    elif disagg is None and 'value' not in gdf_pnt.columns:
+        gdf_pnt['value'] = 1
+
+    return gdf_pnt
+
+
+def disagg_gdf_avg(gdf_pnts):
```

Comment:
I'd suggest a slightly different naming, since the gdf was already disaggregated in its geometry ("rasterized" as you call it) at that point.
E.g. `disagg_vals_avg`

Reply:
Or 'disagg_gdf_values_avg`?

Reply:
also good.